src/main/java/ru/l1ttleO/boyfriend/Utils.java

```diff
@@ -16,81 +28,195 @@ public static String getBeep() {
+            return five_to_zero;
         if (amount % 10 == 1)
-            return first;
-        return second;
+            return one;
+        return two_to_four;
     }
 
-    public static final int @NotNull [] DURATION_MULTIPLIERS = {60, 60, 24, 7, 1};
-    public static final String @NotNull [] @NotNull [] DURATION_TEXTS =
-        {{"s", "секунда", "секунду", "секунды", "секунд"}, {"m", "минута", "минуту", "минуты", "минут"},
-        {"h", "час", "час", "часа", "часов"}, {"d", "день", "день", "дня", "дней"},
-        {"w", "неделя", "неделю", "недели", "недель"}};
+    public static final LinkedHashMap<@NotNull Integer, Pair<@NotNull String, @NotNull String[]>> DURATION_TYPES = new LinkedHashMap<>();
+    
+    static {
+        DURATION_TYPES.put(Calendar.YEAR, Pair.of("y", new String[]{"год", "год", "года", "лет"}));
+        DURATION_TYPES.put(Calendar.MONTH, Pair.of("M", new String[]{"месяц", "месяц", "месяца", "месяцев"}));
+        DURATION_TYPES.put(Calendar.WEEK_OF_YEAR, Pair.of("w", new String[]{"неделя", "неделю", "недели", "недель"}));
+        DURATION_TYPES.put(Calendar.DAY_OF_YEAR, Pair.of("d", new String[]{"день", "день", "дня", "дней"}));
+        DURATION_TYPES.put(Calendar.HOUR_OF_DAY, Pair.of("h", new String[]{"час", "час", "часа", "часов"}));
+        DURATION_TYPES.put(Calendar.MINUTE, Pair.of("m", new String[]{"минута", "минуту", "минуты", "минут"}));
+        DURATION_TYPES.put(Calendar.SECOND, Pair.of("s", new String[]{"секунда", "секунду", "секунды", "секунд"}));
+    }
+    
+    public static final int[] CHECK_ORDER = {Calendar.YEAR, Calendar.MONTH, Calendar.DAY_OF_YEAR,
+        Calendar.SECOND, Calendar.MINUTE, Calendar.HOUR_OF_DAY};
 
-    public static @NotNull String getDurationText(int seconds, final boolean vin) {
-        final StringBuilder out = new StringBuilder();
-        int concatLength = 0;
-        int amount;
-        for (int i = 0; i < DURATION_TEXTS.length; i++) {
-            amount = i < DURATION_TEXTS.length - 1 ? seconds % DURATION_MULTIPLIERS[i] : seconds;
-            if (amount != 0) {
-                switch (concatLength) {
-                    case 0:
+    public static @NotNull String getDurationText(long millis, long from, final boolean accusative) {
+        if (from == 0)
+            from = System.currentTimeMillis();
+        final LinkedHashMap<Integer, Integer> durations = new LinkedHashMap<>();
+        final boolean negative = millis < 0;
+        if (negative)
+            millis *= -1;
+
+        // magic
+        int millisOfTheDay = (int) (millis % (1000 * 60 * 60 * 24));
+        millis -= millisOfTheDay;
+        final GregorianCalendar dateFrom = new GregorianCalendar();
+        dateFrom.setTimeInMillis(negative ? from - millis : from);
+        final GregorianCalendar dateTo = new GregorianCalendar();
+        dateTo.setTimeInMillis(negative ? from : from + millis);
+        millisOfTheDay = Math.round(((float) millisOfTheDay) / 1000);
```

Comment:
```suggestion
        millisOfTheDay = Math.round((float) millisOfTheDay / 1000);
```
`Parentheses around '((float) millisOfTheDay)' are unnecessary`